src/physics/convection.jl

```diff
@@ -202,16 +202,6 @@ function pseudo_adiabat!(
 
     while buoyant && k > 1                  # calculate moist adiabat while buoyant till top
         k -= 1                              # one level up
-            
-        if !saturated                       # if not saturated yet follow dry adiabat
```

Comment:
Ouh, moving this into a simpler `if saturated ... else ... end` structure is not the same though! Starting with `saturated = false` in the first pass saturated might be immediately set to true in which case for that same layer the `if saturated` is being executed. I need to revisit in what order the algorithm is actually supposed to execute. What was the problem here that made you change this?